docs/build-speed.md

```diff
@@ -0,0 +1,200 @@
+
+You can verify that it works using the `which` command:
+
+```
+$ which gcc
+/usr/local/bin/gcc
+```
+
+If the results is `/usr/local/bin/gcc`, then you're effectively calling `ccache` which will wrap the `gcc` calls.
+
+You can then do two Android clean builds. You will notice that the second build was way faster than the first one (it should take seconds rather than minutes).
+While building, you can verify that `ccache` works correctly and check the cache hits/miss rate `ccache -s`
+
+```
+$ ccache -s
+Summary:
+  Hits:             196 /  3068 (6.39 %)
+    Direct:           0 /  3068 (0.00 %)
+    Preprocessed:   196 /  3068 (6.39 %)
+  Misses:          2872
+    Direct:        3068
+    Preprocessed:  2872
+  Uncacheable:        1
+Primary storage:
+  Hits:             196 /  6136 (3.19 %)
+  Misses:          5940
+  Cache size (GB): 0.60 / 20.00 (3.00 %)
+```
+
+Should you need to wipe your cache, you can do so with `ccache --clear`
+
+#### Using this approach on a CI
+
+Ccache uses the `/Users/$USER/Library/Caches/ccache` folder on macOS to store the cache.
+Therefore you could save & restore this folder also on CI to speedup your builds.
+
+However, there are a couple of things to be aware:
+
+1. On CI, we recommend to do a full clean build, to avoid poisoned cache problems. If you follow the approach mentioned in the previous paragraph, you should be able to parallelize the native build on 4 different ABIs and you will most likely not need `ccache` on CI.
+
+2. `ccache` relies on timestamps to compute a cache hit. This doesn't work well on CI as files are re-downloaded at every CI run. To overcome this, you'll need to use the `compiler_check content` option which relies instead on [hashing the content of the file](https://ccache.dev/manual/4.3.html).
+
+### Distributed caches
+
+Similar to local caches, you might want to consider using a distributed cache for your native builds.
+This could be specifically useful in bigger organizations that are doing frequent native builds.
+
+We recommend to use [sccache](https://github.com/mozilla/sccache) to achieve this.
+We defer to the sccache [distributed compilation quickstart](https://github.com/mozilla/sccache/blob/main/docs/DistributedQuickstart.md) for instructions on how to setup and use this tool.
+
+## iOS
+
+### Use a compiler cache
+
+Compilation of Objective-C/C++ and Swift files may also be accelerated by a compiler cache.
+
+Similar to the Android compiler cache section above, we recommend using `ccache`, and you should install it in the same way.
+
+For Xcode to take advantage of a compiler cache three steps are required:
+
+1. You must make symbolic links to the compilers that xcodebuild needs, similar to the links made above for `gcc` and `g++` but Xcode uses `clang` and `clang++`:
+
+```
+ln -s ccache /usr/local/bin/clang
+ln -s ccache /usr/local/bin/clang++
```

Comment:
Agree, though this is it's a user's risk. The brew problem is out of scope in this document. If we're afraid of users creating symlinks, we can update the .mk/CMake files to offer users to prepend `ccache` to all the compilation commands.

That solution however, would be android-only.

Reply:
I wonder if it's enough to leave a sentence saying something like "this may break other projects, use at own risk". But I think people will open issues regardless… 😛